### PR TITLE
ARTEMIS-2453 Fixing deadLock between destroyQueue and removeAddressInfo

### PR DESCRIPTION
**Pull request by clebertsuconic**

### CODE REVIEW

File: artemis-server/src/main/java/org/apache/activemq/artemis/core/postoffice/impl/PostOfficeImpl.java [hunk 1]
```diff
@@ -649,9 +647,7 @@ public AddressInfo removeAddressInfo(SimpleString address, boolean force) throws
 
    @Override
    public AddressInfo getAddressInfo(SimpleString addressName) {
-      synchronized (addressLock) {
-         return addressManager.getAddressInfo(addressName);
-      }
+      return addressManager.getAddressInfo(addressName);
```
Comment (clebertsuconic) [inline artemis-server/src/main/java/org/apache/activemq/artemis/core/postoffice/impl/PostOfficeImpl.java]: There's no need to lock here?

Comment (clebertsuconic) [reply artemis-server/src/main/java/org/apache/activemq/artemis/core/postoffice/impl/PostOfficeImpl.java]: @jbertram What do you think about this?

Comment (jbertram) [reply artemis-server/src/main/java/org/apache/activemq/artemis/core/postoffice/impl/PostOfficeImpl.java]: I don't see a need for the `synchronized (addressLock)` given that the underlying data structure in the address manager is a `ConcurrentHashMap` and no other operations are being done (unlike in the other places which use `addressLock`). I think this fix looks good.

Comment (clebertsuconic) [reply artemis-server/src/main/java/org/apache/activemq/artemis/core/postoffice/impl/PostOfficeImpl.java]: @jbertram thanks.. that confirms what I thought.

I'm running the whole testsuite to confirm it.

File: artemis-server/src/main/java/org/apache/activemq/artemis/core/postoffice/impl/PostOfficeImpl.java [hunk 2]
```diff
@@ -484,7 +482,7 @@ public QueueBinding updateQueue(SimpleString name,
                                    Long delayBeforeDispatch,
                                    SimpleString user,
                                    Boolean configurationManaged) throws Exception {
-      synchronized (addressLock) {
+      synchronized (this) {
```
Comment (michaelandrepearce) [inline artemis-server/src/main/java/org/apache/activemq/artemis/core/postoffice/impl/PostOfficeImpl.java]: Why not make synchronized method for these methods if the whole block is syncd?

**Pull request merged**